File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/signus/KeyForDidTest.java
package org.hyperledger.indy.sdk.signus;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithPoolAndSingleWallet;
import org.hyperledger.indy.sdk.InvalidStateException;
import org.hyperledger.indy.sdk.ledger.Ledger;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;


public class KeyForDidTest extends IndyIntegrationTestWithPoolAndSingleWallet {

	@Test
	public void testKeyForDidWorksForMyDid() throws Exception {
		SignusResults.CreateAndStoreMyDidResult result = Signus.createAndStoreMyDid(wallet, "{}").get();
		String did = result.getDid();
		String key = result.getVerkey();

		String receivedKey = Signus.keyForDid(pool, wallet, did).get();
		assertEquals(key, receivedKey);
	}

	@Test
	public void testKeyForDidWorksForTheirDid() throws Exception {
		Signus.storeTheirDid(wallet, String.format("{\"did\":\"%s\", \"verkey\":\"%s\"}", DID_FOR_MY1_SEED, VERKEY_FOR_MY1_SEED)).get();

		String receivedKey = Signus.keyForDid(pool, wallet, DID_FOR_MY1_SEED).get();
		assertEquals(VERKEY_FOR_MY1_SEED, receivedKey);
	}

	@Test
	public void testKeyForDidWorksForGetKeyFromLedger() throws Exception {
		SignusResults.CreateAndStoreMyDidResult result = Signus.createAndStoreMyDid(this.wallet, TRUSTEE_IDENTITY_JSON).get();
		String trusteeDid = result.getDid();

		Signus.storeTheirDid(wallet, String.format("{\"did\":\"%s\", \"verkey\":\"%s\"}",  DID_FOR_MY1_SEED, VERKEY_FOR_MY1_SEED)).get();
		String nymRequest = Ledger.buildNymRequest(trusteeDid,  DID_FOR_MY1_SEED, VERKEY_FOR_MY1_SEED, null, null).get();
		Ledger.signAndSubmitRequest(pool, wallet, trusteeDid, nymRequest).get();

		String receivedKey = Signus.keyForDid(pool, wallet, DID_FOR_MY1_SEED).get();
		assertEquals(VERKEY_FOR_MY1_SEED, receivedKey);
	}

	@Test
	public void testKeyForDidWorksForNoKey() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStateException.class));

		Signus.keyForDid(pool, wallet, DID_FOR_MY2_SEED).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/SignusTests/KeyForDidTest.cs
using Hyperledger.Indy.LedgerApi;
using Hyperledger.Indy.SignusApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.SignusTests
{

    [TestClass]
    public class KeyForDidTest : IndyIntegrationTestWithPoolAndSingleWallet
    {
        [TestMethod]
        public async Task TestKeyForDidWorksForMyDid()
        {
            var result = await Signus.CreateAndStoreMyDidAsync(wallet, "{}");
            var did = result.Did;
            var key = result.VerKey;

            var receivedKey = await Signus.KeyForDidAsync(pool, wallet, did);

            Assert.AreEqual(key, receivedKey);
        }

        [TestMethod]
        public async Task TestKeyForDidWorksForTheirDid()
        {
            var identityJson = string.Format("{{\"did\":\"{0}\", \"verkey\":\"{1}\"}}", DID_FOR_MY1_SEED, VERKEY_FOR_MY1_SEED);
            await Signus.StoreTheirDidAsync(wallet, identityJson);

            var receivedKey = await Signus.KeyForDidAsync(pool, wallet, DID_FOR_MY1_SEED);

            Assert.AreEqual(VERKEY_FOR_MY1_SEED, receivedKey);
        }

        [TestMethod]
        public async Task TestKeyForDidWorksForGetKeyFromLedger()
        {
            var result = await Signus.CreateAndStoreMyDidAsync(wallet, TRUSTEE_IDENTITY_JSON);
            var trusteeDid = result.Did;

            var identityJson = string.Format("{{\"did\":\"{0}\", \"verkey\":\"{1}\"}}", DID_FOR_MY1_SEED, VERKEY_FOR_MY1_SEED);
            await Signus.StoreTheirDidAsync(wallet, identityJson);

            var nymRequest = await Ledger.BuildNymRequestAsync(trusteeDid, DID_FOR_MY1_SEED, VERKEY_FOR_MY1_SEED, null, null);
            await Ledger.SignAndSubmitRequestAsync(pool, wallet, trusteeDid, nymRequest);

            var receivedKey = await Signus.KeyForDidAsync(pool, wallet, DID_FOR_MY1_SEED);

            Assert.AreEqual(VERKEY_FOR_MY1_SEED, receivedKey);
        }

        [TestMethod]
        public async Task TestKeyForDidWorksForNoKey()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStateException>(() =>
               Signus.KeyForDidAsync(pool, wallet, DID_FOR_MY2_SEED)
           );
        }
    }
}
